src/cmd/sanity_test/sanity_tests.go

```diff
@@ -112,6 +119,76 @@ func checkEmailRestoration(client *msgraphsdk.GraphServiceClient, testUser, fold
 				*restore.GetTotalItemCount())
 			os.Exit(1)
 		}
+
+		checkAllSubFolder(user, restore, "", messageCount)
+	}
+}
+
+func getAllSubFolder(
+	user *users.UserItemRequestBuilder,
```

Comment:
Accepting a user id here and building a call from that will result in less abstraction (smaller mental model) and allow us to shift these calls to an api layer more easily in the future.